kymatio/scattering1d/scattering1d.py

```diff
@@ -233,36 +233,42 @@ def build(self):
         self.phi_f = phi_f
         self._type(torch.FloatTensor)
 
-    def _type(self, target_type):
-        """Change the datatype of the filters
-
-        This function is used internally to convert the filters. It does not
-        need to be called explicitly.
-
-        Parameters
-        ----------
-        target_type : type
-            The desired type of the filters, typically `torch.FloatTensor`
-            or `torch.cuda.FloatTensor`.
+    def _apply(self, fn):
         """
-        cast_psi(self.psi1_f, target_type)
-        cast_psi(self.psi2_f, target_type)
-        cast_phi(self.phi_f, target_type)
+            Mimicates the behavior of the function _apply() of a nn.Module()
```

Comment:
*Mimics

Can't you just register `self.psi` etc as parameters and it would do it automatically?

I guess you would need to inherit from `nn.Module` and I guess we are trying to avoid that?

Reply:
Absolutely correct. This is for a next PR.